pyproject.toml

```diff
@@ -1,10 +1,127 @@
 [build-system]
-requires = ["jupyter_packaging~=0.7.9", "jupyterlab>=3.0.0rc13,==3.*", "setuptools>=40.8.0", "wheel"]
-build-backend = "setuptools.build_meta"
+requires = [
+    "hatchling>=1.3.1",
+    "hatch-jupyter-builder",
+    "jupyterlab>=3.5,<4",
+]
+build-backend = "hatchling.build"
+
+[project]
+name = "jupyter-fs"
+description = "A Filesystem-like mult-contents manager backend for Jupyter"
+version = "0.4.0"
+readme = "README.md"
+license = { file = "LICENSE" }
+requires-python = ">=3.7"
+authors = [
+    { name = "Tim Paine", email = "t.paine154@gmail.com" },
+]
+keywords = [
+    "Jupyter",
+    "JupyterLab",
+    "Extensions",
+    "IPython",
+    "Filesystem",
+    "Tree",
+    "Filetree",
+]
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Framework :: Jupyter",
+    "Framework :: Jupyter :: JupyterLab",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "License :: OSI Approved :: Apache Software License",
+]
+dependencies = [
+    "fs>=2.4.11",
+    "fs-s3fs>=1.1.1",
+    "fs.smbfs>=0.6.3",
+    "jupyterlab>=3.5,<4",
+    "jupyter_server>=2,<3",
```

Comment:
I would want to keep these at the old version for the next release cycle, and then bump them directly after.

Reply:
"fs>=2.4.11",
    "fs-s3fs>=1.1.1",
    "fs.smbfs>=0.6.3",
    "jupyterlab>=3.0.0",
    "jupyter_server>=1.8.0",

Reply:
I don't think the server upgrade is strictly necessary, but the jupyterlab scoping is now that 4 is out